### PR TITLE
Merge the concurrency-5.5-abi-2 branch into release/5.5

### PR DESCRIPTION
**Pull request by rjmccall**
Also includes a cherry-pick of #37060, which was meant to go into the 5.5 branch.

### CODE REVIEW

**Comment by rjmccall** [comment]
@swift-ci Please test

**Comment by rjmccall** [comment]
@swift-ci Please test Windows

**Comment by rjmccall** [comment]
Asking for reviews from the contributors who made functionality changes on this branch

**Review by eeckstein — APPROVED** [review]
lgtm for my changes

**Comment by rjmccall** [comment]
@swift-ci Please test

File: lib/AST/Builtins.cpp [hunk 1]
```diff
@@ -1481,6 +1481,38 @@ static ValueDecl *getResumeContinuationThrowing(ASTContext &ctx,
                             _void);
 }
 
+static ValueDecl *getStartAsyncLet(ASTContext &ctx, Identifier id) {
+//  return getBuiltinFunction(ctx, id, _thin,
+//                            _generics(_unrestricted),
+//                            _parameters(_rawPointer, <function>), TODO: seems we can't express function here?
+//                            _rawPointer)
```
Comment (ktoso) [inline lib/AST/Builtins.cpp:1488]: whoops, commented out code can be removed

**Review by ktoso — APPROVED** [review]
LGTM looks good for the asynclet, task and locals changes 👍

**Review by fredriss — APPROVED** [review]

File: lib/AST/Builtins.cpp (hunk 1 above)
Comment (rjmccall) [reply lib/AST/Builtins.cpp:1488]: I'll let you handle that if you want.

**Pull request merged**